apps/pay/app/[username]/transaction/page.tsx

```diff
@@ -77,11 +80,35 @@ function TransactionsPage() {
   const [transactions, setTransactions] = useState<Transaction[]>([])
   const [cursor, setCursor] = useState<string | null>(null)
   const [hasMore, setHasMore] = useState<boolean>(false)
+  const [posCompanionInstalled, setPosCompanionInstalled] = useState<boolean>(false)
+  const session = useSession()
 
   const { data, loading, error, fetchMore } = useGetPaginatedTransactionsQuery({
     variables: { first: 10 },
   })
 
+  useEffect(() => {
+    const checkRelatedApps = async () => {
+      try {
+        if ("getInstalledRelatedApps" in navigator) {
+          /* eslint @typescript-eslint/ban-ts-comment: "off" */
+          // @ts-ignore-next-line no-implicit-any error
+          const apps = await navigator.getInstalledRelatedApps()
+          const isAppInstalled = apps.some(
+            (app: { id: string }) => app.id === "com.blink.pos.companion",
+          )
+          setPosCompanionInstalled(isAppInstalled)
+        } else {
+          console.error("getInstalledRelatedApps is not supported in this browser.")
```

Comment:
what is the purpose on this? dont we want to inform to the user what is happening?

Reply:
The purpose is to detect if the POS companion app is installed and show the print button if it is installed.